rfcs/013-release_deployment_tracking/README.md

```diff
@@ -4,396 +4,211 @@
+We have multiple projects that are themselves composed of multiple services, which can deploy into multiple environments (e.g. production/staging).
 
-Deploying container images to production at present moves all container images forward to the latest built version.
+Folk are regularly making changes, building updated services and deploying them into the appropriate environments. Keeping track of what is deployed where, by who and why is essential to working effectively.
 
-## Problem statement
+In general the build/release/deployment process can be described as follows:
 
-We wish to track multiple environments with container images at different versions, and have an audit trail for deployments to those environments.
+![Simple overview](simple_flow.png)
 
-A deployment mechanism to take a container image for a service within a project and run that in a given environment is deliberately not described here.
+More specifically including a high level view of infrastructure includes:
 
-SSM parameters are heavily favoured as they provide a data source for terraform.
+- A build environment to build artifacts, e.g. Travis
+- An artifact store to store things, e.g. ECR
+- A database to keep track of what should be released where e.g. DynamoDB
+- A Deployment environment to run things in, e.g. ECS
 
-## Proposed Solution
+![Infrastructure overvier](high_level_infra.png)
 
-This solution describes a set of steps to track build, release and deployment but is deliberately agnostic of deployment mechanism. In addition this RFC **does not** describe a mechanism for reporting build or deployment status.
+### Glossary
 
-### Build artifacts
+In order that we can talk about the numerous concepts flying about here, we'll define some terms and visualise how they fit together:
 
-> **Build artifact:** A container image encapsulating application code that can be configured by environment variable. The artifact _does not_ contain configuration.
+- **project:** A high level abstraction, consisting of one or more **service set**'s. Practically this might indicate whole product and should be a single git repository, e.g. the "catalogue" project.
 
-When container images intended for release are built they will be added to an ECR repository in AWS.
+- **service set:** A functional grouping of **services** within a project. e.g. all the services for the catalogue pipeline. You can have multiple per project, for example in the catalogue project, you've got pipeline, api and adapters.
 
-#### Build artefacts SSM Parameters
+- **service:** Performs a distinct function within a **project**, practically it may be composed of a few closely related containers. e.g. id_minter, requests API, Front-end content app.
 
-The URI of the container image will then be used to update an SSM parameter with a key:
+- **environment:** Where you deploy your **services** when you want them to run! e.g. staging, production
```

Comment:
This definition could equally apply to "ECS or Kubernetes" – it would be good to explain that environments should look broadly similar, but serve different purposes, e.g. prod is for live traffic, staging is for testing.